### ISSUE TITLE
rescue EOFError # short files while using the gem

### ISSUE THREAD

**Issue by Mubarra-naz** [tilo/smarter_csv#217]
I'm using gem in a CLI script, with ruby version 2.4.1

Gem installation is successful, but when I try to use it I get error: 
````
/Users/macbook/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:54:in `require': /Users/macbook/.rvm/gems/ruby-2.4.1/gems/smarter_csv-1.7.4/lib/smarter_csv.rb:549: syntax error, unexpected keyword_rescue, expecting keyword_end (SyntaxError)
      rescue EOFError # short files
            ^
/Users/macbook/.rvm/gems/ruby-2.4.1/gems/smarter_csv-1.7.4/lib/smarter_csv.rb:558: syntax error, unexpected keyword_end, expecting end-of-input
        from /Users/macbook/.rvm/rubies/ruby-2.4.1/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from csv_analysis_script.rb:16:in `<main>'
````

**Comment by tom-lord** [comment]
The library should probably be updated to specify a **minimum ruby version** constraint, because that syntax is dependant on ruby version 2.5+.

Indeed, [it's only being tested against version 2.5+](https://github.com/tilo/smarter_csv/blob/7555a12db857b98e5bdd354b08b43e0b7461090a/.github/workflows/ruby.yml#L23-L28) (although there's also an [old CI configuration file specifying version version 2.2](https://github.com/tilo/smarter_csv/blob/7555a12db857b98e5bdd354b08b43e0b7461090a/.travis.yml#L9-L16)).

**Comment by tom-lord** [comment]
I _think_ support for the older ruby versions may have been lost in [this change](https://github.com/tilo/smarter_csv/pull/194). So you might be able to use version `<= 1.6.1` of this library with ruby v2.4?

**Comment by tilo** [comment]
@tom-lord thank you for reporting this! 

do you have a sample file where this is happening, or do you see the error for any CSV file when running this Ruby version?

**Comment by tilo** [comment]
could it be that there is an issue with an uneven number of quote characters in the CSV file?

**Comment by tilo** [comment]
TL;DR: it is a syntax issue, because Ruby 2.5 started to allow `rescue` clauses [without additional do / end blocks](https://www.ruby-lang.org/en/news/2017/12/25/ruby-2-5-0-released/).

@Mubarra-naz @tom-lord Ruby 2.4 is end-of-life since April 2020

I would recommend to you upgrade to at least 2.7, and then to 3.0.
Ruby 2.7 is also end-of-life end of March this year!

Check out the "Table of Versions" on Wikipedia for which versions are actively supported:
https://en.wikipedia.org/wiki/Ruby_(programming_language)

**Comment by Mubarra-naz** [comment]
Alright. Thanks for the response. It's working with next versions.

**Comment by tom-lord** [comment]
@tilo 

> TL;DR: it is a syntax issue, because Ruby 2.5 started to allow rescue clauses [without additional do / end blocks](https://www.ruby-lang.org/en/news/2017/12/25/ruby-2-5-0-released/).

Yep, this is exactly what I was eluding to in my previous comment :)

To be clear, I'm not personally running into this problem (all applications I currently work on are using ruby 3.x); I was just explaining the cause of the issue.

My comment was that, given this gem is no longer tested (nor works) correctly on ruby <2.5, it ideally ought to [specify a minimum ruby version](https://guides.rubygems.org/specification-reference/#required_ruby_version) in the [gemspec](https://github.com/tilo/smarter_csv/blob/main/smarter_csv.gemspec). That way, the requirement is self-documenting and nobody can accidentally install an incompatible version of the `smarter_csv` library against their older ruby version.

**Comment by tilo** [comment]
if someone is seeing this behavior, you could manually add a begin/end block like this:

```
